### ISSUE TITLE
Make the Select icons and placeholders more accessible  

### ISSUE THREAD

**Issue by haideralsh** [nulogy/design-system#1444]
### Describe the bug

The Select and Async Select use grey/white text/background combination that doesn't meet our accessibility standards. 

### Steps to reproduce

N/A

### Expected behaviour

Use midgrey/white for all the non-decorative icons and text

### Is your team blocked from moving forward by the bug?

No

### Who would you like to fix the bug?

- [ ] Myself
- [X] DesignOps
